### PR TITLE
GROOVY-7932: generate bridge methods for private constructors during …

### PR DESCRIPTION
**Pull request by shils**
…static compilation

### CODE REVIEW

File: src/main/org/codehaus/groovy/transform/sc/StaticCompilationVisitor.java [hunk 1]
```diff
@@ -284,17 +286,36 @@ private static void addPrivateBridgeMethods(final ClassNode node) {
...
-                MethodNode bridge = node.addMethod(
-                        "access$"+i, access,
-                        correctToGenericsSpecRecurse(genericsSpec, method.getReturnType(), methodSpecificGenerics),
-                        newParams,
-                        method.getExceptions(),
-                        returnStatement);
+
+                MethodNode bridge;
+                if (method instanceof ConstructorNode) {
+                    // create constructor with an nested class as the first parameter, creating one if necessary
```
Comment (paulk-asert) [inline src/main/org/codehaus/groovy/transform/sc/StaticCompilationVisitor.java]: trivial typo: an nested => a nested

**Comment by paulk-asert** [comment]
+1, nice work

File: src/main/org/codehaus/groovy/transform/sc/StaticCompilationVisitor.java (hunk 1 above)
Comment (shils) [inline src/main/org/codehaus/groovy/transform/sc/StaticCompilationVisitor.java]: good catch, fixed

**Comment by jwagenleitner** [comment]
+1, I tested and this also fixes [GROOVY-6468](https://issues.apache.org/jira/browse/GROOVY-6468).

**Pull request merged**